server/src/main/java/org/apache/iotdb/db/engine/storagegroup/StorageGroupProcessor.java

```diff
@@ -1440,15 +1439,26 @@ private void logDeletion(long startTime, long endTime, PartialPath deviceId, Str
     }
   }
 
+  private boolean canSkipDelete(TsFileResource tsFileResource, Deletion deletion)
+      throws MetadataException {
+    for (PartialPath p : IoTDB.metaManager.getAllTimeseriesPath(deletion.getPath())) {
```

Comment:
This may cause OOM, you'd better provide a match method for TsFileResource.
For each device in TsFileResource, check if it matches the given path